src/marqo/api/models/embed_request.py

```diff
@@ -0,0 +1,53 @@
+"""Classes used for API communication
+
+Choices (enum-type structure) in fastAPI:
+https://pydantic-docs.helpmanual.io/usage/types/#enums-and-choices
+"""
+import pydantic
+from pydantic import BaseModel, root_validator, Field
+from typing import Union, List, Dict, Optional, Any
+
+from marqo.tensor_search import validation
+from marqo.tensor_search.enums import SearchMethod
+from marqo.tensor_search.models.private_models import ModelAuth
+from marqo.tensor_search.models.score_modifiers_object import ScoreModifier
+from marqo.tensor_search.models.search import SearchContext, SearchContextTensor
+from marqo.tensor_search.models.api_models import BaseMarqoModel
+from marqo.api.exceptions import InvalidArgError
+from marqo.core.models.marqo_index import MarqoIndex
+
+
+class EmbeddingRequest(BaseMarqoModel):
+    # content can be a single query or list of queries. Queries can be a string or a dictionary.
+    content: Union[Union[str, Dict[str, float]], List[Union[str, Dict[str, float]]]]
```

Comment:
Can a list have both strings and dict? I don't think that would be valid. I think what you mean is `Union[str, Dict[str, float], List[str], List[Dict[str, float]]]`

Reply:
A list can have both strings and dict, and it should work fine. So I think the original type signature is correct.

Reply:
Then `Union[str, Dict[str, float], List[Union[str, Dict[str, float]]]]`?